core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/list/NearValueNearbyValueSelector.java

```diff
@@ -0,0 +1,262 @@
+package org.optaplanner.core.impl.heuristic.selector.move.generic.list;
+
+import java.util.Iterator;
+import java.util.Objects;
+
+import org.optaplanner.core.impl.domain.variable.descriptor.GenuineVariableDescriptor;
+import org.optaplanner.core.impl.heuristic.selector.common.iterator.SelectionIterator;
+import org.optaplanner.core.impl.heuristic.selector.common.nearby.NearbyDistanceMatrix;
+import org.optaplanner.core.impl.heuristic.selector.common.nearby.NearbyDistanceMeter;
+import org.optaplanner.core.impl.heuristic.selector.common.nearby.NearbyRandom;
+import org.optaplanner.core.impl.heuristic.selector.value.AbstractValueSelector;
+import org.optaplanner.core.impl.heuristic.selector.value.EntityIndependentValueSelector;
+import org.optaplanner.core.impl.heuristic.selector.value.mimic.MimicReplayingValueSelector;
+import org.optaplanner.core.impl.phase.scope.AbstractPhaseScope;
+import org.optaplanner.core.impl.solver.scope.SolverScope;
+import org.optaplanner.core.impl.util.MemoizingSupply;
+
+public final class NearValueNearbyValueSelector<Solution_> extends AbstractValueSelector<Solution_>
+        implements EntityIndependentValueSelector<Solution_> {
+
+    private final EntityIndependentValueSelector<Solution_> childValueSelector;
+    private final MimicReplayingValueSelector<Solution_> replayingOriginValueSelector;
+    private final NearbyDistanceMeter<?, ?> nearbyDistanceMeter;
+    private final NearbyRandom nearbyRandom;
+    private final boolean randomSelection;
+    // TODO deactivate me when appropriate; consider if this field needs to be included in selector equality
+    private final ListValueNearbyDistanceMatrixDemand<Solution_, ?, ?> nearbyDistanceMatrixDemand;
```

Comment:
I'd argue that the demand is a transitive property of other fields of this class.
Therefore it'll be enough if these fields are used in equals/hashCode.

Reply:
Sorry, the todo is related to something else that has been removed:
https://github.com/kiegroup/optaplanner/blob/6d84dcc5f5b6ff6b9219f773c0ab7e9946d0cca7/core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/selector/entity/nearby/NearEntityNearbyEntitySelector.java#L27-L28